lib/index.js

```diff
@@ -68,12 +70,23 @@ exports.parse = function (definition, options) {
             for (let j = 0; j < opts.length; ++j) {
 
                 if (last) {
-                    errors.push(internals.formatError('Invalid option:', last.name, 'missing value'));
+                    errors.push(internals.formatError('Invalid option:', last.def.name, 'missing value'));
                     continue;
                 }
 
                 const opt = opts[j];
-                const def = keys[opt];
+
+                let jsonDef;
+
+                if (opt.includes('.')) {
+                    const maybeDef = keys[opt.split('.')[0]];
```

Comment:
This does not work with 'json' objects defined at any depth but 0. Eg. for a definition like this:
```js
'pet.owner': { type: 'json'  }
```

I don't know if that should supported, but I don't see any docs, tests, or checks against it.

Reply:
Supporting this causes some usability and predictability issues (esp. with `Bossy.object()`), so it is not supported at this time.  It's enforced by Joi, and `Bossy.object()` has an assertion that relates to this issue.  The test for it is here: https://github.com/hapijs/bossy/pull/77/files#diff-5bb8db779819ddef5956a5d9d5949c05ef7445237656ca37bf2f02720271440bR465-R474